components/style/properties.mako.rs

```diff
@@ -5724,6 +5724,97 @@ pub struct PropertyDeclarationBlock {
     pub normal: Arc<Vec<PropertyDeclaration>>,
 }
 
+impl PropertyDeclarationBlock {
+    // https://drafts.csswg.org/cssom/#serialize-a-css-declaration-block
+    pub fn serialize(&self) -> String {
+
+        let mut result_list = String::new();
+        let mut already_serialized = HashSet::new();
+
+        for declaration in self.normal.iter().chain(self.important.iter()) {
+            let property = declaration.name().to_string();
+            if already_serialized.contains(&property) {
+                continue;
+            }
+
+
+            let mut shorthands = Vec::from(declaration.shorthands());
+
+            if shorthands.len() > 0 {
+                shorthands.sort_by(|a,b| a.longhands().len().cmp(&b.longhands().len()));
+
+                // TODO: Help -  I do not want to do iter cloned, is there a way to
+                // somehow collection into a list or Vec of references?
```

Comment:
What you probably want is a [slice](https://doc.rust-lang.org/stable/book/primitive-types.html#slices).